test/e2e/mock-response-data/snaps/update-snap-binary.ts

```diff
@@ -0,0 +1,214 @@
+import fs from 'fs';
+import path from 'path';
+import { valid as semverValid } from 'semver';
+import { getLocalSnapLatestVersion } from './snap-binary-mocks';
+
+// ANSI escape codes for colors
+const RESET = '\x1b[0m';
+const GREEN = '\x1b[32m';
+const YELLOW = '\x1b[33m';
+const CYAN = '\x1b[36m';
+const RED = '\x1b[31m';
+
+const SNAP_DIR = path.join(__dirname, 'snap-binaries-and-headers');
+
+type Arguments = {
+  snapIdentifier?: string;
+  help?: boolean;
+};
+
+const getArgs = (): Arguments => {
+  const args = process.argv.slice(2);
+  const result: Arguments = {};
+  for (const arg of args) {
+    if (arg.startsWith('--')) {
+      const rawArg = arg.substring(2);
+      if (rawArg === 'help') {
+        result.help = true;
+      } else if (rawArg.includes('@')) {
+        result.snapIdentifier = rawArg;
+      } else {
+        console.warn(
+          `${YELLOW}Warning: Unrecognized argument format: ${arg}${RESET}`,
+        );
+      }
+    }
+  }
+  return result;
+};
+
+const fetchSnapData = async (
+  snapName: string,
+  version: string,
+): Promise<{ headers: Record<string, string>; body: Buffer } | undefined> => {
+  const url = `https://registry.npmjs.org/@metamask/${snapName}/-/${snapName}-${version}.tgz`;
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      if (response.status === 404) {
+        console.error(
+          `${RED}Snap not found in the registry: ${snapName}@${version}${RESET}`,
+        );
+        process.exit(1);
+        return undefined;
+      }
+      throw new Error(`Failed to fetch snap: ${response.statusText}`);
+    }
+    const headers = Object.fromEntries(response.headers.entries()) as Record<
+      string,
+      string
+    >;
+    const arrayBuffer = await response.arrayBuffer();
+    const body = Buffer.from(arrayBuffer);
+    return { headers, body };
+  } catch (error) {
+    console.error(`${RED}Error fetching snap data:${RESET}`, error);
+    process.exit(1);
+    return undefined;
+  }
+};
+
+const saveSnapFiles = (
+  snapName: string,
+  version: string,
+  data: { headers: Record<string, string>; body: Buffer },
+) => {
+  const txtFilePath = path.join(SNAP_DIR, `${snapName}@${version}.txt`);
+  const headersFilePath = path.join(
+    SNAP_DIR,
+    `${snapName}@${version}-headers.json`,
```

Comment:
:information_source:  file names updated to follow the suggested format @Mrtenz :pray: